File: testdata/testkit/csharp/Adapter/InMemoryRepo.cs
using Microsoft.Extensions.Logging;
using TestKit.Domain;

namespace TestKit.Adapter;

public class InMemoryRepo : IRepository
{
    private readonly Dictionary<string, Entity> _store = new();
    private readonly ILogger<InMemoryRepo> _logger;

    public InMemoryRepo(ILogger<InMemoryRepo> logger) => _logger = logger;

    public Task<Entity?> FindByIdAsync(string id)
    {
        _logger.LogDebug("Finding {Id}", id);
        _store.TryGetValue(id, out var entity);
        return Task.FromResult(entity);
    }

    public Task SaveAsync(Entity entity)
    {
        _logger.LogInformation("Saving {Id}", entity.Id);
        _store[entity.Id] = entity;
        return Task.CompletedTask;
    }
}

File: testdata/testkit/kotlin/src/main/kotlin/adapter/InMemoryRepo.kt
package adapter

import domain.Entity
import domain.Repository
import kotlinx.coroutines.sync.Mutex
import kotlinx.coroutines.sync.withLock

class InMemoryRepo : Repository {
    private val store = mutableMapOf<String, Entity>()
    private val mutex = Mutex()

    override suspend fun findById(id: String): Entity? = mutex.withLock { store[id] }
    override suspend fun save(entity: Entity) = mutex.withLock { store[entity.id] = entity }
}
